src/main/anovos/data_ingest/data_sampling.py

```diff
@@ -0,0 +1,68 @@
+from pyspark.sql.functions import concat, lit
+from anovos.shared.utils import attributeType_segregation
+
+
+def data_sample(
+    idf,
+    list_of_cols="all",
+    drop_cols=[],
+    fraction=0.1,
+    method_type="random",
+    stratified_type="proportionate",
+    seed_value=12,
+    unique_threshold=0.5,
+):
+    if type(fraction) != float and type(fraction) != int:
+        raise TypeError("Invalid input for fraction")
+    if fraction < 0 or fraction > 1:
+        raise TypeError("Invalid input for fraction. Fraction value is between 0 and 1")
+    if type(seed_value) != int:
+        raise TypeError("Invalid input for seed_value")
+    if method_type not in ["stratified", "random"]:
+        raise TypeError("Invalid input for data_sample method_type")
+    if method_type == "stratified":
+        if stratified_type not in ["proportionate", "disproportionate"]:
```

Comment:
I would suggest using terms such as "population" & "balanced", "population" meaning stratified sampling outcome follows the population distribution and "balanced" meaning stratified sampling  output follows the balanced distribution based on the strata columns